logic/src/parameters/boolean.rs

```diff
@@ -0,0 +1,28 @@
+use serde::{Deserialize, Serialize};
+use std::str::FromStr;
+
+use super::{populator::RandomNumberGenerator, GenerateRandomValues, ParameterError};
+
+#[derive(Serialize, Deserialize, Debug, Default)]
```

Comment:
IntParameter and StringParameter don't derive Default. I confirmed this builds

```suggestion
#[derive(Serialize, Deserialize, Debug)]
```